_posts/2024-12-25-elm-after-7-years.md

```diff
@@ -0,0 +1,35 @@
+date:   2024-12-25 12:00:00 -0500
+categories: blog
+author: Aaron
+author_twitter: shamshirz
+image: /assets/elm/elm.webp
+---
+My development perspective on using Elm, and when it will serve you. [![Elm-lang logo]({{ "/assets/elm/elm.webp" | relative_url }}){:class="excerpt-image"}](https://elm-lang.org)
+
+**Note about Shamshirz**
+
+*This post reflects my experience working extensively with [Elm](https://elm-lang.org) at a [B2B insurance startup](http://corvusinsurance.com/) from 2017 to acquisition in 2024, as the 2nd Engineer.*
+
+I'm looking at a new year of big changes and I want to share my appreciation and advocacy through reflection. Elm is my Gold Standard, best in class UI development kit. I had experience with SSR templating languages, vanilla JS sites, and React just prior to picking up Elm. Elm was my introduction to a pleasant type systems (👎 Java). The type system is friendly, clarifies thinking, and it provides the best feedback loop I’ve experienced in engineering. I learned that Elm can be rocket fuel and I hope to articulate the the conditions for success that we encountered. Elm isn’t without its limitations, but allow me to convince you why it's worth it, dear reader.
+
+## Why Elm Shines
+The type system is simple and delightful. The compiler is fast, capable, and descriptive enough to be a coding assistant without the need of an LLM. Problem solving is mostly modeling types to reflect your needs and the code writes itself. The biggest hurdle is adapting to the syntax the flow of data. The language is small! Once you are past basics, you can participate in any discussion and I always found it fun to iterate, learn, and improve. We even found that it was unnecessary to write tests. 
+
+Running Elm in production genuinely changed how I saw front end development. The compiler’s feedback was the end of the loop, rather than manual UI testing after a change. The compiler will assert whether you’ve covered every possible case and handled every operation that could fail. The implications are enormous and easily overlooked. No debugging in the browser locally, in Staging, in Production, or trying to reproduce a problem a user saw in their browser. It’s very hard to truly appreciate the lack of something found in many little moments, but the overall impact was profound. I didn’t view UI work as tedious and error prone. The only negative that I experienced was that Elm became “boring”, it didn’t need me. I had complete confidence that Elm could guide new engineers to success and adding my experience wouldn’t bring any added benefit. I subscribe to Dan McKinley's ideal of [“boring technology”](https://mcfunley.com/choose-boring-technology) being a good thing that allows you to put energy elsewhere. We know when Elm shines and we know it's failure modes quite well. The important decision to make is whether you are willing to accept the trade that Elm requires in order to benefit.
```

Comment:
It's interesting that you state that Elm is now a "boring technology". I think to the avg reader that might be surprising - the perception might still be that it's niche / not as tested as other "boring" technologies. may be worth some kind of "I know, surprising! But imo it is boring now" or smth like that